packages/dataviews/src/dataviews-layouts/picker-table/index.tsx

```diff
@@ -0,0 +1,499 @@
+									}
+									scope="col"
+								>
+									<ColumnHeaderMenu
+										ref={ headerMenuRef( column, index ) }
+										fieldId={ column }
+										view={ view }
+										fields={ fields }
+										onChangeView={ onChangeView }
+										onHide={ onHide }
+										setOpenedFilter={ setOpenedFilter }
+										canMove={
+											view.layout?.enableMoving ?? true
+										}
+									/>
+								</th>
+							);
+						} ) }
+					</tr>
+				</thead>
+				{ /* Render grouped data if groupByField is specified */ }
+				{ hasData && groupField && dataByGroup ? (
+					<Composite
+						virtualFocus
+						orientation="vertical"
+						render={ ( { children } ) => <>{ children }</> }
+					>
+						{ Array.from( dataByGroup.entries() ).map(
+							( [ groupName, groupItems ] ) => (
+								<tbody
+									key={ `group-${ groupName }` }
+									role="group"
+								>
+									<tr
+										className="dataviews-view-table__group-header-row"
+										role="presentation"
+									>
+										<td
+											colSpan={
+												columns.length +
+												( hasPrimaryColumn ? 1 : 0 ) +
+												1
+											}
+											className="dataviews-view-table__group-header-cell"
+											role="presentation"
+										>
+											{ sprintf(
+												// translators: 1: The label of the field e.g. "Date". 2: The value of the field, e.g.: "May 2022".
+												__( '%1$s: %2$s' ),
+												groupField.label,
+												groupName
+											) }
+										</td>
+									</tr>
+									{ groupItems.map( ( item, index ) => (
+										<TableRow
+											key={ getItemId( item ) }
+											item={ item }
+											fields={ fields }
+											id={
+												getItemId( item ) ||
+												index.toString()
+											}
+											view={ view }
+											titleField={ titleField }
+											mediaField={ mediaField }
+											descriptionField={
+												descriptionField
+											}
+											selection={ selection }
+											getItemId={ getItemId }
+											onChangeSelection={
+												onChangeSelection
+											}
+											multiselect={ isMultiselect }
+										/>
+									) ) }
+								</tbody>
+							)
+						) }
+					</Composite>
+				) : (
+					<Composite
+						render={ <tbody role="presentation" /> }
+						virtualFocus
+						orientation="vertical"
+					>
+						{ hasData &&
+							data.map( ( item, index ) => (
+								<TableRow
+									key={ getItemId( item ) }
+									item={ item }
+									fields={ fields }
+									id={ getItemId( item ) || index.toString() }
+									view={ view }
+									titleField={ titleField }
+									mediaField={ mediaField }
+									descriptionField={ descriptionField }
+									selection={ selection }
+									getItemId={ getItemId }
+									onChangeSelection={ onChangeSelection }
+									multiselect={ isMultiselect }
+									posinset={
+										isInfiniteScroll ? index + 1 : undefined
```

Comment:
Same here as the previous comment. `posinset` can be passed for the `listbox`'s `option` elements too.